### PR TITLE
Convert shared/constants/hardware-wallets.js -> Typescript

### PR DESCRIPTION
**Pull request by ryanml**
## Explanation
Part of Level 34 of our Typescript conversion project: https://github.com/MetaMask/metamask-extension/issues/17213

As part of this refactor, `shared/constants/keyrings.js` has been absorbed in to `shared/constants/hardware-wallets.js`

## Manual Testing Steps
1. Ensure automated tests pass
2. Perform a general smoke test of hardware features and connectivity

## Pre-merge author checklist

- [X] I've clearly explained:
  - [X] What problem this PR is solving
  - [X] How this problem was solved
  - [X] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Review by brad-decker — APPROVED** [review]
LGTM good work

File: shared/constants/hardware-wallets.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+export enum HardwareTransportStates {
+  none = 'NONE',
+  verified = 'VERIFIED',
+  deviceOpenFailure = 'DEVICE_OPEN_FAILURE',
+  unknownFailure = 'UNKNOWN_FAILURE',
+}
+
+export enum HardwareAffiliateLinks {
+  ledger = 'https://shop.ledger.com/?r=17c4991a03fa',
+  gridplus = 'https://gridplus.io/?afmc=7p',
```
Comment (DDDDDanica) [inline shared/constants/hardware-wallets.ts:38]: should we transform them into camelCase?

**Review by DDDDDanica — APPROVED** [review]

File: shared/constants/hardware-wallets.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+ * keyring types. Both simple and HD are treated as default but we do special
+ * case accounts managed by a hardware wallet.
+ */
+export enum HardwareKeyringTypes {
+  ledger = 'Ledger Hardware',
+  trezor = 'Trezor Hardware',
+  lattice = 'Lattice Hardware',
+  qr = 'QR Hardware Wallet Device',
+  hdKeyTree = 'HD Key Tree',
+  imported = 'Simple Key Pair',
```
Comment (Gudahtt) [inline shared/constants/hardware-wallets.ts:12]: Why were these moved here? This is now very misleading. These last two are not hardware wallets.
